### PR TITLE
ipykernel 6.15.2

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Changelog: **no** breaking changes https://github.com/ipython/ipykernel/blob/v6.15.2/CHANGELOG.md
License: https://github.com/ipython/ipykernel/blob/v6.15.2/COPYING.md
Requirements: https://github.com/ipython/ipykernel/blob/v6.15.2/pyproject.toml

Actions:
1. Update scripts
2. Update host and run dependencies
3. Update test dependencies and pinnings
4. Add license_url

### CODE REVIEW

**Comment by skupr-anaconda** [comment]
CI issues on Prefect for linux platforms. The test_shutdown_subprocesses is failing has to do with shutting down the system using process groups. We put the entire build into a process group. It looks that the test is checking to see that a child process or process group is shutdown after it’s requested to, but it’s not.
We should skip test_shutdown_subprocesses.

**Comment by skupr-anaconda** [comment]
I opened a new PR for v6.15.2 https://github.com/AnacondaRecipes/ipykernel-feedstock/pull/23

**Comment by skupr-anaconda** [comment]
The bug on `win64`:
```
______________________ test_matplotlib_inline_on_import _______________________

    def test_matplotlib_inline_on_import():
        pytest.importorskip("matplotlib", reason="this test requires matplotlib")
        with kernel() as kc:
            cell = "\n".join(
                ["import matplotlib, matplotlib.pyplot as plt", "backend = matplotlib.get_backend()"]
            )
            _, reply = execute(cell, user_expressions={"backend": "backend"}, kc=kc)
            _check_status(reply)
            backend_bundle = reply["user_expressions"]["backend"]
            _check_status(backend_bundle)
>           assert "backend_inline" in backend_bundle["data"]["text/plain"]
E           assert 'backend_inline' in "'agg'"

..\_test_env\lib\site-packages\ipykernel\tests\test_kernel.py:327: AssertionError
=========================== short test summary info ===========================
FAILED ..\_test_env\lib\site-packages\ipykernel\tests\test_kernel.py::test_matplotlib_inline_on_import
```

 was fixed by cleaning up the windows builders on concourse I guess

File: recipe/meta.yaml [hunk 1]
```diff
@@ -26,32 +24,37 @@ build:
...
+    - jupyter_client >=6.1.12
+    - matplotlib-inline >=0.1.0
     - nest-asyncio
-    - python
-    - tornado >=4.2,<7.0
-    - traitlets >=5.1.0,<6.0
+    - packaging
+    - psutil
+    - pyzmq >=17
+    - tornado >=6.1
```
Comment (katietz) [inline recipe/meta.yaml]: is it save to give up upper bounds here?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -26,32 +24,37 @@ build:
...
+    - traitlets >=5.1.0
 
 test:
   requires:
     - curio  # [not win]
     - flaky
-    - numpy >=1.16.1  # introduced numpy.core._multiarray_umath
+    - jupyter_client
+    - numpy
```
Comment (katietz) [inline recipe/meta.yaml]: is there no more the need for _multiarray_umath?  Not sure if this constrain on numpy should be dropped

File: recipe/run_test.py [hunk 3]
```diff
@@ -56,7 +59,17 @@ def go():
     skips = ["flaky"]
 
     if len(skips) == 1:
-        pytest_args += ["-k", "not {}".format(*skips)]
+        # 2022/8/29: CI issues on Prefect for linux platforms:
```
Comment (katietz) [inline recipe/run_test.py:62]: nice finding ... but well, pretty CI centric fix

**Review by katietz — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: I've added the upper bound again for `tornado` and `traitlets`

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: You are right. It's better not to drop the constraint.

**Review by varlackc — APPROVED** [review]
Other than the issue with the `run_test.py` file being `ci` centric everything else seems to be fine.

**Comment by skupr-anaconda** [comment]
@katietz I've addressed your comments and updated matplotlib-inline to 0.1.6

**Pull request merged**